charts/seed-bootstrap/charts/etcd-druid/templates/etcd-crd.yaml

```diff
@@ -0,0 +1,1317 @@
+                    which delta snapshots will be taken
+                  type: string
+                deltaSnapshotPeriod:
+                  description: DeltaSnapshotPeriod defines the period after which
+                    delta snapshots will be taken
+                  type: string
+                fullSnapshotSchedule:
+                  description: FullSnapshotSchedule defines the cron standard schedule
+                    for full snapshots.
+                  type: string
+                garbageCollectionPeriod:
+                  description: GarbageCollectionPeriod defines the period for garbage
+                    collecting old backups
+                  type: string
+                garbageCollectionPolicy:
+                  description: GarbageCollectionPolicy defines the policy for garbage
+                    collecting old backups
+                  enum:
+                  - Exponential
+                  - LimitBased
+                  type: string
+                port:
+                  description: Port define the port on which etcd-backup-restore server
+                    will exposed.
+                  type: integer
+                resources:
+                  description: 'Resources defines the compute Resources required by
+                    backup-restore container. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/'
+                  properties:
+                    limits:
+                      additionalProperties:
+                        type: string
+                      description: 'Limits describes the maximum amount of compute
+                        resources allowed. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/'
+                      type: object
+                    requests:
+                      additionalProperties:
+                        type: string
+                      description: 'Requests describes the minimum amount of compute
+                        resources required. If Requests is omitted for a container,
+                        it defaults to Limits if that is explicitly specified, otherwise
+                        to an implementation-defined value. More info: https://kubernetes.io/docs/concepts/configuration/manage-compute-resources-container/'
+                      type: object
+                  type: object
+                store:
+                  description: Store defines the specification of object store provider
+                    for storing backups.
+                  properties:
+                    container:
+                      type: string
+                    prefix:
+                      type: string
+                    provider:
+                      enum:
+                      - aws
+                      - gcp
+                      - os
+                      - azure
+                      - alicloud
```

Comment:
Nice catch @vpnachev. Let's please get rid of this list in g/g repo. We don't want to maintain such an enum. Also, I recommend to rename `os` to `openstack` like it is done with the gardener-extensions. `os` is ambigious.

Reply:
What about this comment?

Reply:
etcd-backup-restore uses objectstore names eg. `s3` while input to druid will be the cloud-provider name eg.`aws`. I have a [switch](https://github.com/gardener/etcd-druid/blob/master/pkg/utils/miscellaneous.go#L147) in druid to handle that case. I kept the enum here so that I dont get rogue values for which I may error out in druid. I can remove this, but at the cost of throwing an error for providers I do not have a mapping in druid.

Reply:
I think that’s fine. The etcd resource is in the seed, so it’s not too dangerous. And even if you get a value you don’t understand then you can throw an appropriate error in the default case of your switch.

Reply:
I think that’s fine. The etcd resource is in the seed, so it’s not too dangerous. And even if you get a value you don’t understand then you can throw an appropriate error in the default case of your switch.